### ISSUE TITLE
Problem: filter api missing events when internal ws reconnect

### ISSUE THREAD

**Issue by yihuang** [crypto-org-chain/cronos#665]
@mmsqe found that the reason for the mempool test failure is due to internal ws connection reconnect occasionally, when that happens, the API lose messages.

**Comment by mmsqe** [comment]
since `No way to detect data loss` based on current [desgin](https://github.com/tendermint/tendermint/blob/master/docs/architecture/adr-075-rpc-subscription.md#issue-2-inconvenient-client-api), not sure if we can deprecate websocket and switch to long poll on events endpoint with after cursor, starting [1st](https://github.com/mmsqe/tendermint/pull/44/files) perf improve

**Comment by yihuang** [comment]
> since `No way to detect data loss` based on current [desgin](https://github.com/tendermint/tendermint/blob/master/docs/architecture/adr-075-rpc-subscription.md#issue-2-inconvenient-client-api), not sure if we can deprecate websocket and switch to long poll on events endpoint with after cursor, starting [1st](https://github.com/mmsqe/tendermint/pull/44/files) perf improve

yeah, that's the plan https://github.com/tendermint/tendermint/issues/9305
